metriken-exposition/src/snapshot.rs

```diff
@@ -59,4 +59,22 @@ impl Snapshot {
             histograms: Vec::new(),
         }
     }
+
+    #[cfg(feature = "json")]
+    pub fn to_json<T>(val: &T) -> Result<Vec<u8>, crate::Error>
```

Comment:
Actually, thinking a little more. What's the point of a wrapper error type here? Why not just return the serde/msgpack error?

Reply:
I felt like it would be nicer to have a wrapped error type that way we're not exposing implementation specific errors. But I'm okay eitherway.